src/components/App/App.jsx

```diff
@@ -1,115 +1,210 @@
 import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import {
-  HashRouter as Router,
+  BrowserRouter as Router,
   Route,
   Redirect,
   Switch,
 } from 'react-router-dom';
 
-import { useDispatch } from 'react-redux';
+// AUTHENTICATION:
+import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
 
+// CUSTOM COMPONENTS:
 import Nav from '../Nav/Nav';
 import Footer from '../Footer/Footer';
-
-import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
-
 import AboutPage from '../AboutPage/AboutPage';
+import Message from '../MessageAll/MessageAll';
+import MessagesList from '../MessageAll/MessagesList';
+// import MessageDetail from '../MessageDetail/MessageDetail';
 import UserPage from '../UserPage/UserPage';
-import InfoPage from '../InfoPage/InfoPage';
 import LandingPage from '../LandingPage/LandingPage';
 import LoginPage from '../LoginPage/LoginPage';
 import RegisterPage from '../RegisterPage/RegisterPage';
+import DiscoverPage from '../DiscoverPage/DiscoverPage';
+import SearchNetwork from '../SearchNetwork/SearchNetwork';
+import StyleGuide from '../StyleGuide/StyleGuide';
+import VendorProfile from '../Profile/VendorProfile';
+import RegisterVendorPage from '../RegisterPage/RegisterVendorPage';
 
 import './App.css';
 
+// Material-UI:
+import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
+import { CssBaseline } from '@material-ui/core';
+
+const contriveTheme = createMuiTheme({
+  palette: {
+    background: {
+      paper: '#fff',
+      default: '#fff',
+      level2: '#f5f5f5',
+      level1: '#fff',
+    },
+    secondary: {
+      light: '#fffff9',
+      main: '#fefcc6',
+      dark: '#cac995',
+      contrastText: '#000000',
+    },
+    primary: {
+      light: '#cd7c50',
+      main: '#984f26',
+      dark: '#652500',
+      contrastText: '#ffffff',
+    },
+  },
+  typography: {
+    fontFamily: ['Forum', 'Raleway'].join(','),
+  },
+});
+
 function App() {
   const dispatch = useDispatch();
 
   useEffect(() => {
     dispatch({ type: 'FETCH_USER' });
+    dispatch({ type: 'FETCH_VENDOR_TYPES' });
+    dispatch({ type: 'FETCH_SPECIAL_FEATURES' });
   }, [dispatch]);
 
   return (
-    <Router>
-      <div>
+    <ThemeProvider theme={contriveTheme}>
+      <CssBaseline />
+      <Router>
         <Nav />
-        <Switch>
-          {/* Visiting localhost:3000 will redirect to localhost:3000/home */}
-          <Redirect exact from="/" to="/home" />
-
-          {/* Visiting localhost:3000/about will show the about page. */}
-          <Route
-            // shows AboutPage at all times (logged in or not)
-            exact
-            path="/about"
-          >
-            <AboutPage />
-          </Route>
-
-          {/* For protected routes, the view could show one of several things on the same route.
+        <div>
+          <Switch>
+<<<<<<< HEAD
```

Comment:
whoops 😱 

This is where even a basic functional test ("can we render <App />" ) would save you some heartache 🐴 

I'm down like a 🤡  if you want a hand setting that up

Reply:
WE WILL TAKE YOU UP ON THAT!!